### ISSUE TITLE
Can't send payload

### ISSUE THREAD

**Issue by akzReekoh** [splunk/splunk-javascript-logging#6]
Here's the code I'm using (got it directly from your npmjs page):

var SplunkLogger = require('splunk-logging').Logger;

var config = {
    token: '<myToken>',
    url: '<myInstance>'
};

var Logger = new SplunkLogger(config);

var payload = {
    // Message can be anything; doesn't have to be an object
    message: {
        temperature: "70F",
        chickenCount: 500
    }
};

console.log("Sending payload", payload);
Logger.send(payload, function(err, resp, body) {
    // If successful, body will be { text: 'Success', code: 0 }
    console.log("Response from Splunk", body);
});

And I always get this error:
/home/adinglasan/Desktop/splunk/node_modules/splunk-logging/splunklogger.js:466
                if (!requestError && body && body.code.toString() !== "0") {
                                                                       ^

TypeError: Cannot read property 'toString' of undefined
    at Request._callback (/home/adinglasan/Desktop/splunk/node_modules/splunk-logging/splunklogger.js:466:55)

**Comment by shakeelmohamed** [comment]
Thanks for the bug report @akzReekoh! Can you try adding `console.log(body);` on line 464 in `splunklogger.js` and let us know the output?

**Comment by akzReekoh** [comment]
Hi,


This is the output:


<!doctype html><html><head><meta http-equiv="content-type" content="text/html; charset=UTF-8"><meta http-equiv="refresh" content="1;url=http://54.174.219.239:8000/en-US/services/collector/event/1.0"><title>303 See Other</title><script type="text/javascript">
var hashTag = '', hashPos = document.location.href.indexOf('#');
if (hashPos > -1) { hashTag = document.location.href.substr(hashPos); }
document.location = "http://54.174.219.239:8000/en-US/services/collector/event/1.0" + hashTag;
</script>
</head><body><h1>See Other</h1><p>The resource has moved temporarily <a href="http://54.174.219.239:8000/en-US/services/collector/event/1.0">here</a>.</p></body></htm


________________________________

**Comment by shakeelmohamed** [comment]
Thanks @akzReekoh, it looks like you're using the splunkweb port (8000 by default) instead of the HTTP Event Collector port (8088 by default). Try changing the port and let us know if that helps

**Comment by calvinlfer** [comment]
Hello @shakeelmohamed 
https://github.com/splunk/splunk-javascript-logging/commit/9cb0149bec06f83301dd2d273e9afad34108152a should be merged in ASAP as it prevents situations where the Splunk Forwarder becomes unresponsive and results in exceptions being thrown due to `body.code` being undefined and the `toString` method is being invoked on an undefined method causing an exception to be thrown. Are there any plans for this?

**Comment by shakeelmohamed** [comment]
Hi @calvinlfer,

What's the response you get when the forwarder is under heavy load?
We'd like to get a release out next week with a fix for #9 also.

**Comment by calvinlfer** [comment]
Hey @shakeelmohamed, we get a type-error exception thrown because the Splunk forwarder fails to return a JSON response under heavy load. So `body` exists but `body.code` is undefined and `toString` is called on `body.code`. I can provide you the exact response shortly

**Comment by hackmad** [comment]
@shakeelmohamed @calvinlfer Sanitized private IDs and URLs but its returning a 504 with some HTML content:

```
{
  "statusCode": 504,
  "body": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01 Transitional//EN\" \"http://www.w3.org/TR/html4/loose.dtd\">\n<HTML><HEAD><META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=iso-8859-1\">\n<TITLE>ERROR: The request could not be satisfied</TITLE>\n</HEAD><BODY>\n<H1>ERROR</H1>\n<H2>The request could not be satisfied.</H2>\n<HR noshade size=\"1px\">\nCloudFront attempted to establish a connection with the origin, but either the attempt failed or the origin closed the connection.\n<BR clear=\"all\">\n<HR noshade size=\"1px\">\n<PRE>\nGenerated by cloudfront (CloudFront)\nRequest ID: ***********\n</PRE>\n<ADDRESS>\n</ADDRESS>\n</BODY></HTML>",
  "headers": {
    "content-type": "text/html",
    "content-length": "669",
    "connection": "close",
    "server": "CloudFront",
    "date": "Thu, 03 Nov 2016 16:41:31 GMT",
    "x-cache": "Error from cloudfront",
    "via": "1.1 "**************" (CloudFront)",
    "x-amz-cf-id": "**************"
  },
  "request": {
    "uri": {
      "protocol": "https:",
      "slashes": true,
      "auth": null,
      "host": "******:443",
      "port": "443",
      "hostname": "******",
      "hash": null,
      "search": null,
      "query": null,
      "pathname": "/services/collector/event/1.0",
      "path": "/services/collector/event/1.0",
      "href": "https://**************:443/services/collector/event/1.0"
    },
    "method": "POST",
    "headers": {
      "Authorization": "Splunk ***********",
      "Content-Type": "application/x-www-form-urlencoded",
      "accept": "application/json",
      "content-length": 181
    }
  }
}
```

**Comment by shakeelmohamed** [comment]
Thanks @hackmad this is very helpful

**Comment by shakeelmohamed** [comment]
We've just released v0.9.2 with this fix. Give it a try!

**Comment by calvinlfer** [comment]
Thanks a lot
